src/Compilers/CSharp/Test/Semantic/Semantics/RefLocalsAndReturnsTests.cs

```diff
@@ -1260,10 +1254,10 @@ void local()
         }
     }
 }");
-            comp.VerifyDiagnostics(
-                // (10,34): error CS8176: Iterators cannot have by-reference locals
-                //                 ref readonly int x = ref (new int[1])[0]; // 1
-                Diagnostic(ErrorCode.ERR_BadIteratorLocalType, "x").WithLocation(10, 34));
+            comp.VerifyEmitDiagnostics(
+                // (12,30): error CS4013: Instance of type 'ref readonly int' cannot be used inside a nested function, query expression, iterator block or async method
```

Comment:
>error CS4013: Instance of type 'ref readonly int' cannot be used inside a nested function, query expression, iterator block or async method [](http://example.com/codeflow?start=28&length=139)

Do we need to adjust this error slightly? It sounds like we couldn't use the `ref readonly int` instance at all in the iterator block but presumably it would be allowed in `yield return x;` if the preceding `yield return 1;` was removed.

Reply:
Yes, thanks. I will also change its name because `ERR_SpecialByRefInLambda` sounds confusing. I will add a prototype comment now to avoid making the diff in this PR large.